### ISSUE TITLE
[FVT] Error message found in CN:/var/log/xcat/xcat.log after CN is booted.

### ISSUE THREAD

**Issue by caomengmeng** [xcat2/xcat-core#1168]
Find below error in CN:/var/log/xcat/xcat.log after install sles12.1 successfully with latest daily build:
```
  <error>Unable to read private ECDSA key from /etc/xcat/hostkeys</error>
  <error>Unable to read private ECDSA key from /etc/xcat/hostkeys</error>
```
Test ENV:
```
[root@c910f result]# rpm -qa | grep -i xCAT
xCAT-server-2.12-snap201605190710.noarch
xCAT-test-2.12-snap201605190711.noarch
syslinux-xcat-3.86-2.noarch
xCAT-genesis-base-x86_64-2.12-snap201605051534.noarch
ipmitool-xcat-1.8.15-1.x86_64
xCAT-client-2.12-snap201605190710.noarch
xCAT-buildkit-2.12-snap201605190711.noarch
xCAT-2.12-snap201605190711.x86_64
elilo-xcat-3.14-4.noarch
conserver-xcat-8.1.16-10.x86_64
grub2-xcat-2.02-0.16.el7.snap201506090204.noarch
perl-xCAT-2.12-snap201605190710.noarch
xCAT-genesis-scripts-x86_64-2.12-snap201605190711.noarch
[root@c910f result]# cat /etc/*release
Red Hat Enterprise Linux Server release 6.7 (Santiago)
Red Hat Enterprise Linux Server release 6.7 (Santiago)
```

**Comment by daniceexi** [comment]
This is not a problem.
The ``ECDSA key`` is not supported if the xCAT MN is a a little old. So in your scenario, the rh6.7 MN does not generate the ``ECDSA`` key by default.

**Comment by daniceexi** [comment]
The possible thing we need to fix is whether the error message can be improved.
@caomengmeng could you paste the error message here?

**Comment by caomengmeng** [comment]
@daniceexi , Ｄo you mean the message in MN?
The message in CN is already in this issue, please see the beginning of first comments.
```
  <error>Unable to read private ECDSA key from /etc/xcat/hostkeys</error>
  <error>Unable to read private ECDSA key from /etc/xcat/hostkeys</error>
```

**Comment by gurevichmark** [comment]
@daniceexi I had to fix something similar for the genesis boot in #888. Same error was being displayed.
The issue was the command `ssh-keygen` inside the `doxcat` script.
When used `ssh-keygen -A` option it generates all the keys and errors will not be displayed. Instead we used
`ssh-keygen -q -t rsa` and `ssh-keygen -q -t dsa` which generated only 2 keys. However, `-A` option is new and is not supported on older versions of `ssh-keygen`

**Comment by daniceexi** [comment]
@gurevichmark I root cause for this issue should be the xCATMN is old (rh6.7) which does not support to  create ``ecdsa`` key by default. ( Or may does not support). Then the postscript ``remoteshell`` would fail to download the ecdsa key. This is a proper behaviour, but we need to refine the error message.

The change should be to dump the error message from ``grep`` to /dev/null. And echo an error message at end of the code section which getting the ecdsa key.

~~~~
244   #check whether the message is an error or not
245   grep -E '<error>' /tmp/ssh_ecdsa_hostkey
~~~~

@gurevichmark , could you make this fix?

**Comment by gurevichmark** [comment]
Ok, assigning to myself to make the fix.

**Comment by gurevichmark** [comment]
@daniceexi remoteshell postscript contains 2 other, similar statements:

`111 grep -E '<error>' /tmp/ssh_dsa_hostkey` and
`177 grep -E '<error>' /tmp/ssh_rsa_hostkey`

To be consistent, should similar changes be made there as you suggested for 
`245   grep -E '<error>' /tmp/ssh_ecdsa_hostkey` ?

**Comment by daniceexi** [comment]
@gurevichmark Yes, I think so.

**Comment by gurevichmark** [comment]
Fixed by pull request #1243

**Comment by caomengmeng** [comment]
Works well and doesn't show the same error
